ompi/mca/coll/base/coll_base_reduce_scatter_block.c

```diff
@@ -511,3 +510,407 @@ ompi_coll_base_reduce_scatter_block_intra_recursivehalving(
         free(tmprecv_raw);
     return err;
 }
+
+/*
+ * ompi_mirror_perm: Retruns mirror permutation of nbits low-order bits
+ *                   of value.
+ */
+static int ompi_mirror_perm(unsigned int value, int nbits)
+{
+    int perm = 0;
+    for (int i = 0; i < nbits; i++) {
```

Comment:
Not that I really think that it is performance critical, but you can implement this much simpler by extracting the last bit from ```value``` and pushing it left on ```perm```.
```
for( int i = 0; i < nbits; i++ ) {
    perm |= (value & 0x1);
    value >>= 1;
    perm <<= 1;
 }
```